Files/Views/LayoutModes/GridViewBrowser.xaml.cs

```diff
@@ -574,9 +574,9 @@ private void FileList_ContainerContentChanging(ListViewBase sender, ContainerCon
                 UninitializeDrag(args.ItemContainer);
                 args.ItemContainer.PointerPressed -= FileListGridItem_PointerPressed;
 
-                if (args.Item is ListedItem item && item.ItemPropertiesInitialized && item.FileImage != null)
+                if (args.Item is ListedItem item && item.ItemPropertiesInitialized && item.LoadFileIcon)
                 {
-                    args.RegisterUpdateCallback(4, (s, c) => ParentShellPageInstance.FilesystemViewModel.UnloadItemThumbnail(item));
+                    ParentShellPageInstance.FilesystemViewModel.UnloadItemThumbnail(item);
```

Comment:
It looks like UnloadItemThumbnail is never called when put inside the RegisterUpdateCallback call.

Reply:
Good catch